### PR TITLE
Delete service accounts

### PR DESCRIPTION
**Pull request by dimakis**
To verify:
- you must have an existing service account
- save this playbook as `run.yml`:
 
``` ansible
---
- name: Test
  hosts: localhost
  gather_facts: true
  connection: local
  tasks:
  - name: Delete service account 
    delete_service_account_by_id:
      service_account_id: "SERVICE_ACCOUNT_ID"
```

run
```shell
ansible-playbook 'run.yml'
```

### CODE REVIEW

File: plugins/modules/delete_service_account_by_id.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+# Pass in a message
+  - name: Delete Service Account
+    delete_service_account_by_id:
+        service_account_id: "service_account_id"
+'''
+
+RETURN = r'''
+# These are examples of possible return values, and in general should use other names for return values.
+original_message:
+    description: The original params that were passed in to be deleted.
```
Comment (wtrocki) [inline plugins/modules/delete_service_account_by_id.py:45]: Not sure what that means. `params` is not an english word

File: plugins/modules/delete_service_account_by_id.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        supports_check_mode=True
+    )
+
+    if module.check_mode:
+        module.exit_json(**result)
+
+    token = auth.get_access_token()
+    
+    configuration = rhoas_service_accounts_mgmt_sdk.Configuration(
+        host = "https://sso.redhat.com/auth/realms/redhat-external",
```
Comment (wtrocki) [inline plugins/modules/delete_service_account_by_id.py:82]: Antipattern. We should not have it hardcoded etc.

File: plugins/modules/delete_service_account_by_id.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    
+    # Enter a context with an instance of the API client
+    with rhoas_service_accounts_mgmt_sdk.ApiClient(configuration) as api_client:
+        api_instance = service_accounts_api.ServiceAccountsApi(api_client)
+        try:
+            service_account_id = module.params['service_account_id']
+            result['original_message'] = f'Deleting Service Account with ID: {service_account_id}'
+            api_instance.delete_service_account(service_account_id)
+            result['changed'] = True
+            result['message'] = "Service Account deleted successfully"
```
Comment (wtrocki) [inline plugins/modules/delete_service_account_by_id.py:95]: Docs says that this is returned only when error happens :P

**Review by wtrocki — APPROVED** [review]
Minor docs issues. LGTM

File: plugins/modules/delete_service_account_by_id.py (hunk 2 above)
Comment (wtrocki) [reply plugins/modules/delete_service_account_by_id.py:82]: No. Having constant used across modules is good

Comment (dimakis) [reply plugins/modules/delete_service_account_by_id.py:82]: but thats not really possible as there are two host addresses, we've got that `sso...` for service accounts and `api.openshift...` for kafka mgmt 🤔

Comment (wtrocki) [reply plugins/modules/delete_service_account_by_id.py:82]: Not possible to have 2 constants?

Comment (wtrocki) [reply plugins/modules/delete_service_account_by_id.py:82]: I'm looking to have central point for all configuration based elements. In future we can seed them from rhoas/config.json for now just 2 constants in one file

**Comment by dimakis** [comment]
@wtrocki all comments addressed 👍

**Pull request merged**